### DISCUSSION TITLE
Can't get the HASP Core functionality automation to work

### DISCUSSION THREAD

**Discussion by jjlink** [HASwitchPlate/HASPone#35] [Q&A] [answered]
On my first unit it said setting up home assistant but it never finished. On my second unit running the  run action does nothing as far as I can see. Any ideas?

**Comment by cboehnke** [comment]
There are much easier instruction now - after an ESP and LCD firmware update to HASPOne v1.0 everything is much faster. 
And Home Assistant integration happens via Blueprints now. 
https://github.com/HASwitchPlate/HASPone/wiki/Updating-legacy-HASP-to-HASPone

Full video walkthrough - and hugely entertaining conversation with Luma on Dr Zzs livestream from 2 weeks agon:
https://www.youtube.com/watch?v=RYuOvhQ0mtQ

**Reply by jjlink** [reply]
Yes, i followed these as shown on Dr Zzs video up to the point  where his switchplate came up with the three pages.  In my case the first one hung at that point, and the second one did nothing. Thanks

**Comment by aderusha** [comment]
> On my first unit it said setting up home assistant but it never finished. On my second unit running the run action does nothing as far as I can see. Any ideas?

What do you see on each unit right now?  Trying to get a sense of where you're at to see if we can't get you moving in the right direction.

**Reply by jjlink** [reply]
Right now they both display the WiFi Connected and the MQTT Connected. The odd part is the automations says this:

"HASP Core functionality plate01
Last triggered: Never
HASP Core functionality plate02
Last triggered: Never"

I tried running both of them multiple times, I also deleted them and re-ran the blueprint to recreate them.

**Reply by aderusha** [reply]
OK, so both showing connections on the display is good news!  If you look under Configuration > Devices you should see both of your devices (looks like you're calling them plate01 and plate02).  Can you open up the plate01 device page and post a screenshot of what you're seeing there?

**Reply by jjlink** [reply]
Ok, I deleted all the swplate entities, automations. Started over and plate2 worked fine, it has page1,2,3 on it now. Plate one says setting up home assistant, so I guess its hung up. I did get two notifications, one for each swplate.

**Comment by jjlink** [comment]
![image](https://user-images.githubusercontent.com/6171474/110838777-b7105a80-8257-11eb-8b80-bf392e89404f.png)
![image](https://user-images.githubusercontent.com/6171474/110838824-c5f70d00-8257-11eb-953b-3341fdc8e3db.png)

**Comment by jjlink** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/6171474/110839326-46b60900-8258-11eb-937b-0e2ddefb75d8.png)
![image](https://user-images.githubusercontent.com/6171474/110839385-559cbb80-8258-11eb-80cd-ebf5ae1e34e6.png)

**Reply by aderusha** [reply]
Allright so `plate02` is looking OK there, `plate01` is missing some bits and is suggesting those bits have been disabled, which is weird.

In the device view for `plate01`, hit the `delete` button to remove all the associated entities, then reboot plate01 and let it come up to the connected screen, then try deploying the blueprint again.

**Reply by jjlink** [reply]
Ok, it mostly working however the paging is a bit odd on plate01. On page 1 and 3 the top rectangle turns black. it does not do that on page 2.

**Reply by jjlink** [reply]
neverminded I rebooted it and its all working fine on both plates. Thanks for your help!

**Reply by aderusha** [reply]
Fantastic, glad it's sorted!
